server/sonar-server/src/main/java/org/sonar/server/component/ComponentService.java

```diff
@@ -124,81 +69,18 @@ public void updateKey(DbSession dbSession, String projectOrModuleKey, String new
     index(component.uuid());
   }
 
+  // TODO should be moved to ComponentUpdater
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)